util/github_util.go

```diff
@@ -0,0 +1,20 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/thedevsaddam/gojsonq/v2"
```

Comment:
To Stephen's point, I think both of these libraries could be replaced with libraries that Go provides:

- encoding/json - https://pkg.go.dev/encoding/json
- net/http - https://pkg.go.dev/net/http

Reply:
The code is just picking a single field that is the latest tag from the json response.I am not sure if creating a struct and then maintaining it may pose some issues in future, in case the json response structure changes. Moreover there would be delay in unmarshalling the response to a struct.
```net/http``` is not being used because ```resty``` is much comprehensive and it provides more features like timeouts, retries etc in a more declarative fashion.

Reply:
for maintainability, i think it would be best to use the go standard libraries. Most go developers are already familiar with the http library and the json library. I'm not sure there is much benefit using these libraries over the standard libraries and we do introduce some code smells such as magic strings with `tag_name` vs using dot notation.

Reply:
Added test cases and use the go standard libraries. Please have a look.
```
Hari.Nadiminti@EUMBQTHV267VXT dynamic-buildkite-template % go run main.go trivy
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 5m0s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config%                                                                                                               
Hari.Nadiminti@EUMBQTHV267VXT dynamic-buildkite-template % go test -v ./...
?       dynamic-buildkite-template      [no test files]
?       dynamic-buildkite-template/cmd  [no test files]
?       dynamic-buildkite-template/generator    [no test files]
?       dynamic-buildkite-template/test [no test files]
=== RUN   Test_GenerateSuccess
--- PASS: Test_GenerateSuccess (0.00s)
=== RUN   Test_GenerateWrongTemplatePath
--- PASS: Test_GenerateWrongTemplatePath (0.00s)
PASS
ok      dynamic-buildkite-template/test/generator_test  (cached)
?       dynamic-buildkite-template/util [no test files]
=== RUN   Test_GetLatestTagSuccess
--- PASS: Test_GetLatestTagSuccess (0.36s)
=== RUN   Test_GetLatestTagError
--- PASS: Test_GetLatestTagError (0.15s)
PASS
ok      dynamic-buildkite-template/test/githubutil_test 1.076s
```